subsys/llext/Kconfig

```diff
@@ -14,7 +14,7 @@ if LLEXT
 choice LLEXT_BINARY_TYPE
 	prompt "Binary object type for llext"
 	default LLEXT_TYPE_ELF_OBJECT if ARM || ARM64
-	default LLEXT_TYPE_ELF_SHAREDLIB if XTENSA
+	default LLEXT_TYPE_ELF_RELOCATABLE if XTENSA
```

Comment:
should we join this with below `XTENSA || RISCV`? BTW, what's the default for ARC? Should there be a fallback default? All the choice options below are `n` by default, so it's rather unclear.

Reply:
Joined with RISCV

I need to cleanup the defaults and the types here in a follow up. This PR is fixing an issue with testing where twister will try and run tests for a target and toolchain that don't work.